From 10a2b10f25cc703d33405d0a32de9797e04205db Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Sat, 8 Aug 2026 12:19:48 +0200
Subject: [PATCH] FIX: Match D-Bus names to the verified Flatpak

Lower-case domain-name in D-Bus names so that it matches
the application ID of the verified Flatpak app on Flathub.

Flatpak'ed application can only own D-Bus names that match their
application ID, so com.bambulab.BambuStudio could not own
com.BambuLab.BambuStudio.

Fixes: a803118282d8 ("Fix DBus name case mismatch breaking Linux single-instance file passing")
---
 src/slic3r/GUI/InstanceCheck.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/slic3r/GUI/InstanceCheck.cpp b/src/slic3r/GUI/InstanceCheck.cpp
index ba8b88aab274..bec86c166036 100644
--- a/src/slic3r/GUI/InstanceCheck.cpp
+++ b/src/slic3r/GUI/InstanceCheck.cpp
@@ -236,10 +236,10 @@ namespace instance_check_internal
 			dbus_uint32_t 	serial = 0;
 			const char* sigval = message_text.c_str();
 			//std::string		interface_name = "com.prusa3d.prusaslicer.InstanceCheck";
-			std::string		interface_name = "com.BambuLab.BambuStudio.InstanceCheck.Object" + version;
+			std::string		interface_name = "com.bambulab.BambuStudio.InstanceCheck.Object" + version;
 			std::string   	method_name = "AnotherInstance";
 			//std::string		object_name = "/com/prusa3d/prusaslicer/InstanceCheck";
-			std::string		object_name = "/com/BambuLab/BambuStudio/InstanceCheck/Object" + version;
+			std::string		object_name = "/com.bambulab/BambuStudio/InstanceCheck/Object" + version;
 
 
 			// initialise the error value
@@ -538,7 +538,7 @@ namespace MessageHandlerDBusInternal
 	        "       <arg name=\"data\" direction=\"out\" type=\"s\" />"
 	        "     </method>"
 	        "   </interface>"
-	        "   <interface name=\"com.BambuLab.BambuStudio.InstanceCheck\">"
+	        "   <interface name=\"com.bambulab.BambuStudio.InstanceCheck\">"
 	        "     <method name=\"AnotherInstance\">"
 	        "       <arg name=\"data\" direction=\"in\" type=\"s\" />"
 	        "     </method>"
@@ -576,7 +576,7 @@ namespace MessageHandlerDBusInternal
 	{
 		const char* interface_name = dbus_message_get_interface(message);
 	    const char* member_name    = dbus_message_get_member(message);
-	    std::string our_interface  = "com.BambuLab.BambuStudio.InstanceCheck.Object" + wxGetApp().get_instance_hash_string();
+	    std::string our_interface  = "com.bambulab.BambuStudio.InstanceCheck.Object" + wxGetApp().get_instance_hash_string();
 	    BOOST_LOG_TRIVIAL(trace) << "DBus message received: interface: " << interface_name << ", member: " << member_name;
 	    if (0 == strcmp("org.freedesktop.DBus.Introspectable", interface_name) && 0 == strcmp("Introspect", member_name)) {
 	        respond_to_introspect(connection, message);
@@ -596,8 +596,8 @@ void OtherInstanceMessageHandler::listen()
     int 				 name_req_val;
     DBusObjectPathVTable vtable;
     std::string 		 instance_hash  = wxGetApp().get_instance_hash_string();
-	std::string			 interface_name = "com.BambuLab.BambuStudio.InstanceCheck.Object" + instance_hash;
-    std::string			 object_name 	= "/com/BambuLab/BambuStudio/InstanceCheck/Object" + instance_hash;
+    std::string			 interface_name = "com.bambulab.BambuStudio.InstanceCheck.Object" + instance_hash;
+    std::string			 object_name 	= "/com.bambulab/BambuStudio/InstanceCheck/Object" + instance_hash;
 
     //BOOST_LOG_TRIVIAL(debug) << "init dbus listen " << interface_name << " " << object_name;
     dbus_error_init(&err);
-- 
2.55.0

